crates/rpc/rpc-api/src/eth_filter.rs

```diff
@@ -15,7 +14,10 @@ pub trait EthFilterApi {
 
     /// Creates a pending transaction filter and returns its id.
     #[method(name = "newPendingTransactionFilter")]
-    async fn new_pending_transaction_filter(&self) -> RpcResult<FilterId>;
+    async fn new_pending_transaction_filter(
+        &self,
+        kind: Option<PendingTransactionFilterKind>,
```

Comment:
we can get rid of the option if we implement Serialize+Deserialize accordingly
```suggestion
        kind: PendingTransactionFilterKind,
```

Reply:
> we can get rid of the option if we implement Serialize+Deserialize accordingly

I was thinking of making it optional to clearly indicate that it's not mandatory. This way, people won't be confused about the type. What are your thoughts?

based on this https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_newpendingtransactionfilter